File: admin-panel/types/portfolio.ts
export interface KillSwitchState {
  active: boolean;
  reason: string | null;
  last_triggered: string;
}

export interface SafeModeState {
  active: boolean;
  level: number;
}

export interface ReadinessState {
  score: number;
  grade: string;
  components: {
    data: number;
    model: number;
    market: number;
  };
}

export interface ConfidenceState {
  score: number;
  trend: "rising" | "stable" | "falling";
  history: number[];
}

export interface SystemSummary {
  kill_switch: KillSwitchState;
  safe_mode: SafeModeState;
  readiness: ReadinessState;
  confidence: ConfidenceState;
  risk_level: "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";
  active_constraints: number;
  portfolio_value: number;
  daily_pnl: number;
  daily_pnl_pct: number;
  open_positions: number;
  agents_active: number;
  last_cycle: string;
}

export interface RiskContributor {
  asset: string;
  risk: number;
  weight: number;
  contribution_pct: number;
}

export interface ScenarioResult {
  name: string;
  probability: number;
  impact: number;
  color: string;
}

export interface MarginalRiskData {
  total_risk: number;
  expected_return: number;
  quality_ratio: number;
  contributors: RiskContributor[];
  scenarios: ScenarioResult[];
}

export interface LearningPattern {
  pattern: string;
  frequency: number;
  confidence: number;
  win_rate: number;
  last_seen: string;
}

export interface Alert {
  id: number;
  type: "info" | "warning" | "error" | "critical";
  message: string;
  timestamp: string;
  dismissed: boolean;
}

export interface AISuggestion {
  key: string;
  current: number;
  suggested: number;
  reason: string;
}

File: admin-api/models/portfolio.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class KillSwitchState(BaseModel):
    active: bool
    reason: str | None
    last_triggered: str


class SafeModeState(BaseModel):
    active: bool
    level: int


class ReadinessComponents(BaseModel):
    data: float
    model: float
    market: float


class ReadinessState(BaseModel):
    score: float
    grade: str
    components: ReadinessComponents


class ConfidenceState(BaseModel):
    score: float
    trend: Literal["rising", "stable", "falling"]
    history: list[float]


class SystemSummary(BaseModel):
    """Matches frontend SystemSummary exactly."""

    kill_switch: KillSwitchState
    safe_mode: SafeModeState
    readiness: ReadinessState
    confidence: ConfidenceState
    risk_level: Literal["LOW", "MEDIUM", "HIGH", "CRITICAL"]
    active_constraints: int
    portfolio_value: float
    daily_pnl: float
    daily_pnl_pct: float
    open_positions: int
    agents_active: int
    last_cycle: str


class RiskContributor(BaseModel):
    asset: str
    risk: float
    weight: float
    contribution_pct: float


class ScenarioResult(BaseModel):
    name: str
    probability: float
    impact: float
    color: str


class MarginalRiskData(BaseModel):
    """Matches frontend MarginalRiskData exactly."""

    total_risk: float
    expected_return: float
    quality_ratio: float
    contributors: list[RiskContributor]
    scenarios: list[ScenarioResult]


class KillSwitchToggle(BaseModel):
    active: bool


class LearningPattern(BaseModel):
    """Matches frontend LearningPattern exactly."""

    pattern: str
    frequency: int
    confidence: float
    win_rate: float
    last_seen: str


class Alert(BaseModel):
    """Matches frontend Alert exactly."""

    id: int
    type: Literal["info", "warning", "error", "critical"]
    message: str
    timestamp: str
    dismissed: bool


class AISuggestion(BaseModel):
    """Matches frontend AISuggestion exactly."""

    key: str
    current: float
    suggested: float
    reason: str
